File: frontend/lib/data/dtos/trusted_contact_dto.dart
import '../../domain/models/trusted_contact/trusted_contact.dart';

/// Mapper responsabile della conversione dei dati tra il formato JSON del backend
/// e l'oggetto di dominio [TrustedContact].
abstract class TrustedContactDTO {
  /// Trasforma un oggetto JSON [data] in un oggetto [TrustedContact].
  static TrustedContact fromJson(Map<String, dynamic> data) {
    return TrustedContact(
      id: (data['contactId'] ?? data['contact_id'])?.toString() ?? '',
      name: (data['name'] ?? data['contact_name'])?.toString() ?? '',
      email: (data['email'] ?? data['contact_email'])?.toString() ?? '',
      phoneNumber: (data['phoneNumber'] ?? data['contact_phone_number'])?.toString() ?? '',
    );
  }

  /// Converte un oggetto [contact] di tipo [TrustedContact] in una mappa JSON.
  static Map<String, dynamic> toJson(TrustedContact contact) {
    return {
      'contactId': contact.id,
      'name': contact.name,
      'email': contact.email,
      'phoneNumber': contact.phoneNumber,
      'contact_id': contact.id,
      'contact_name': contact.name,
      'contact_email': contact.email,
      'contact_phone_number': contact.phoneNumber,
    };
  }
}

File: backend/src/trusted_contact/domain/dtos/trusted_contact_dto.py
from dataclasses import dataclass

@dataclass
class TrustedContactDTO:
    user_id: str
    contact_id: str
    contact_name: str
    contact_email: str
    contact_phone_number: str

    def to_dict(self) -> dict:
        return {
            "user_id": self.user_id,
            "contact_id": self.contact_id,
            "contact_name": self.contact_name,
            "contact_email": self.contact_email,
            "contact_phone_number": self.contact_phone_number
        }
    
    @staticmethod
    def from_domain(trusted_contact):
        return TrustedContactDTO(
            user_id=trusted_contact.user_id,
            contact_id=trusted_contact.contact_id,
            contact_name=trusted_contact.contact_name,
            contact_email=trusted_contact.contact_email,
            contact_phone_number=trusted_contact.contact_phone_number
        )